kornia/filters/bilateral.py

```diff
@@ -1,13 +1,64 @@
 from __future__ import annotations
 
 from kornia.core import Module, Tensor, pad
-from kornia.core.check import KORNIA_CHECK_IS_TENSOR, KORNIA_CHECK_SHAPE
+from kornia.core.check import KORNIA_CHECK, KORNIA_CHECK_IS_TENSOR, KORNIA_CHECK_SHAPE
 
-from .gaussian import get_gaussian_kernel2d
-from .kernels import _unpack_2d_ks
+from .kernels import _unpack_2d_ks, get_gaussian_kernel2d
 from .median import _compute_zero_padding
 
 
+def _bilateral_blur(
+    input: Tensor,
+    guidance: Tensor | None,
+    kernel_size: tuple[int, int] | int,
+    sigma_color: float | Tensor,
+    sigma_space: tuple[float, float] | Tensor,
+    border_type: str = 'reflect',
+    color_distance_type: str = "l1",
+) -> Tensor:
+    "Single implementation for both Bilateral Filter and Joint Bilateral Filter"
+
+    KORNIA_CHECK_IS_TENSOR(input)
```

Comment:
Is tensor is included in check shape

Reply:
From what I see, currently `CHECK_SHAPE()` does not have `IS_TENSOR()` check

Reply:
Ok, that changed. Fair enough